src/coreclr/jit/objectalloc.cpp

```diff
@@ -1335,3 +1483,1734 @@ void ObjectAllocator::RewriteUses()
+    // single-def and the def dominates all the Ui uses, but this may not work out...?)
+    //
+    // Also we do not expect any Ti or Ui use to be a GDV guard. U's typically arise from
+    // inlining under a successful GDV of V, and should have exact types, resolving any
+    // potential GDV in the inlinee.
+    //
+    // First, find the one and only def of V (aka `defBlock`).
+    //
+    EnumeratorVar* v      = nullptr;
+    bool const     foundV = info->m_appearanceMap->Lookup(info->m_local, &v);
+
+    if (!foundV)
+    {
+        JITDUMP("Unexpected: no appearance info for V%02u\n", info->m_local);
+        return false;
+    }
+
+    if (v->m_hasMultipleDefs)
+    {
+        JITDUMP("Unexpected: V%02u multiply defined\n", info->m_local);
+        return false;
+    }
+
+    BasicBlock* const defBlock = v->m_def->m_block;
+    Statement* const  defStmt  = v->m_def->m_stmt;
+
+    JITDUMP("V%02u has single def in " FMT_BB " at [%06u]\n", info->m_local, defBlock->bbNum,
+            comp->dspTreeID(defStmt->GetRootNode()));
+
+    // We expect to be able to follow all paths from alloc block to defBlock
+    // without reaching "beyond" defBlock.
+    //
+    // Because we are inside a GDV hammock, we do not expect to see a normal
+    // flow path from allocBlock that can bypass defBlock. For now we trust
+    // that is the case.
+    //
+    // toVisit: blocks we need to visit to determine extent of cloning
+    // visited: block we will need to clone
+    // toVisitTryEntry: subset of above that are try entries.
+    //
+    CompAllocator                alloc(comp->getAllocator(CMK_ObjectAllocator));
+    ArrayStack<BasicBlock*>      toVisit(alloc);
+    jitstd::vector<BasicBlock*>* visited         = new (alloc) jitstd::vector<BasicBlock*>(alloc);
+    jitstd::vector<BasicBlock*>* toVisitTryEntry = new (alloc) jitstd::vector<BasicBlock*>(alloc);
+
+    BitVecTraits traits(comp->compBasicBlockID, comp);
+    BitVec       visitedBlocks(BitVecOps::MakeEmpty(&traits));
+    toVisit.Push(allocBlock);
+
+    // todo -- some kind of runaway size limit
+    //
+    while (toVisit.Height() > 0)
+    {
+        BasicBlock* const visitBlock = toVisit.Pop();
+        if (!BitVecOps::TryAddElemD(&traits, visitedBlocks, visitBlock->bbID))
+        {
+            continue;
+        }
+
+        if (visitBlock != allocBlock)
+        {
+            visited->push_back(visitBlock);
+        }
+
+        // We expect this stretch of blocks to all be in the same EH region.
+        //
+        if (!BasicBlock::sameEHRegion(allocBlock, visitBlock))
+        {
+            JITDUMP("Unexpected: new EH region at " FMT_BB "\n", visitBlock->bbNum);
+            return false;
+        }
+
+        if (visitBlock == defBlock)
+        {
+            continue;
+        }
+
+        JITDUMP("walking through " FMT_BB "\n", visitBlock->bbNum);
+
+        for (BasicBlock* const succ : visitBlock->Succs())
```

Comment:
`VisitRegularSuccs` would be better to use here

Reply:
Should this actually use `VisitAllSuccs`? It seems this needs to take into account any possible control flow path if it's trying to validate some form of post dominance.

Reply:
See note below, since we're in an inlined GDV region, and we're validing all blocks are in the same EH region, any EH successor would bypass assignment and so would not flow into the cloned code we're going to create.

Reply:
Makes sense.